src/snowflake/cli/plugins/nativeapp/manager.py

```diff
@@ -310,6 +350,25 @@ def sync_deploy_root_with_stage(self, role: str) -> DiffResult:
             % self.deploy_root
         )
         diff: DiffResult = stage_diff(self.deploy_root, self.stage_fqn)
+
+        # If we are syncing specific files/directories, remove everything else from the diff
+        if len(paths_to_sync) > 0:
+            resolved_paths_to_sync = [resolve_without_follow(p) for p in paths_to_sync]
+            if not recursive:
+                _verify_no_directories(resolved_paths_to_sync)
```

Comment:
shouldn't this apply even when paths to sync isn't empty? At this point, the CLI options are irrelevant, we should honour the arguments that were passed to this call without any assumptions.

Reply:
You mean apply when the paths _are_ empty? (full deploy)
Yes, added this verification in `build_bundle()` too

Reply:
Oops, yeah, I said the opposite of what I meant.